src/operators/mpo.jl

```diff
@@ -277,25 +277,18 @@ function Base.:*(mpo1::InfiniteMPO, mpo2::InfiniteMPO)
 
     T = promote_type(scalartype(mpo1), scalartype(mpo2))
     make_fuser(i) = fuser(T, left_virtualspace(mpo2, i), left_virtualspace(mpo1, i))
-    fusers = PeriodicArray(map(make_fuser, 1:L))
 
-    Os = map(1:L) do i
-        return _fuse_mpo_mpo(mpo1[i], mpo2[i], fusers[i], fusers[i + 1])
+    Os = map(zip(parent(mpo1), parent(mpo2))) do (O1, O2)
+        return fuse_mul_mpo(O1, O2)
     end
     return InfiniteMPO(Os)
 end
 
-function _fuse_mpo_mpo(O1::MPOTensor, O2::MPOTensor, Fₗ, Fᵣ)
-    return @plansor O′[-1 -2; -3 -4] := Fₗ[-1; 1 4] *
-                                        O2[1 2; -3 3] *
-                                        O1[4 -2; 2 5] *
-                                        conj(Fᵣ[-4; 3 5])
-end
-
 # TODO: I think the fastest order is to start from both ends, and take the overlap at the
 # largest virtual space cut, but it might be better to just multithread both sides and meet
 # in the middle
-function TensorKit.dot(bra::FiniteMPS{T}, mpo::FiniteMPO, ket::FiniteMPS{T}) where {T}
+function TensorKit.dot(bra::FiniteMPS{T}, mpo::FiniteMPO{<:MPOTensor},
```

Comment:
I guess if you replace the `similar(...)` calls with the correct `allocate_GL`/`allocate_GR`, this is again agnostic to the eltype?

Reply:
I only found one `similar` I could replace (I also think the `bra` and `ket` were switched in the `similar` call there, so I think that's a fortunate accidental fix). Did you have any others in mind that I maybe missed?

Reply:
Yeah sorry my comment was badly placed. You got the one I meant, sadly we don't currently have a `isomorphism!` implementation so the other `rho` instantiations will have to remain as they are.